src/types.ts

```diff
@@ -6,9 +6,24 @@
  */
 
 import { Config } from './config';
+import Transaction from './models/transaction';
 import Input from './models/input';
 import FullNodeConnection from './new/connection';
 
+export interface IInputSignature {
+  inputIndex: number,
+  addressIndex: number,
+  signature: Buffer,
+  pubkey: Buffer,
+}
+
+/**
+ * Signature that receives the data to sign.
```

Comment:
Yes, the initial implementation would receive the data to sign and the address index.
During the implementation I changed since this is not ideal, because it would require retrieving the key from storage and decrypting + signing it for each input instead of fetching once and signing multiple times.

This would be bad specially for HSM since we would need to start a session for each address key, even when we have 2 inputs for the same address.

Reply:
I'll fix the docstring